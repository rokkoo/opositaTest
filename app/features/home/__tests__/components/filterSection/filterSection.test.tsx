import React from 'react';
import { fireEvent, screen } from '@testing-library/react-native';
import FilterSection from '@app/features/home/components/filterSection';
import { renderWithReactQuery } from '@app/helpers/tests/renderWithReactQuery';
import { AppTheProvider } from '@app/helpers/tests/appTheProvider';

describe('FilterSection', () => {
  test('it should render correctly', () => {
    renderWithReactQuery(
      <AppTheProvider>
        <FilterSection />
      </AppTheProvider>,
    );

    expect(screen).toMatchSnapshot();

    const sectionTitle = screen.queryByText('Filtra por un nombre');
    expect(sectionTitle).toBeTruthy();
  });

  test('it should update input value', async () => {
    const bookTitle = 'A Game Of Thrones';

    renderWithReactQuery(
      <AppTheProvider>
        <FilterSection />
      </AppTheProvider>,
    );

    const input = await screen.findByTestId('input');
    fireEvent.changeText(input, bookTitle);

    expect(input.props.value).toBe(bookTitle);
  });
});
