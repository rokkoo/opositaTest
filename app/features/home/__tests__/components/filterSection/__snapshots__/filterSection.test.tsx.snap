// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterSection it should render correctly 1`] = `
<View
  style={
    {
      "paddingVertical": 16,
    }
  }
  testID="filter-section"
>
  <Text
    style={
      [
        {
          "color": "black",
          "fontFamily": "System",
        },
        {
          "color": "black",
          "fontSize": 16,
          "fontWeight": "bold",
        },
        undefined,
      ]
    }
  >
    Filtra por un nombre
  </Text>
  <View
    style={
      [
        {
          "paddingTop": 16,
        },
      ]
    }
  />
  <TextInput
    onChangeText={[Function]}
    placeholder="A Game of Thrones"
    placeholderTextColor="#0D0D0D"
    style={
      {
        "borderColor": "#9D9D9D",
        "borderRadius": 4,
        "borderWidth": 3,
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
    testID="input"
    value=""
  />
</View>
`;
